src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/shadow_labyrinth.cpp

```diff
@@ -189,6 +189,57 @@ void instance_shadow_labyrinth::OnCreatureRespawn(Creature* creature)
     }
 }
 
+// The Screaming Hall 
+// All Worldstates get Activated via database when door opens
+void instance_shadow_labyrinth::OnCreatureGroupDespawn(CreatureGroup* group, Creature* /*pCreature*/)
+{
+
+    switch (group->GetGroupId())
+    {
+        // Blackheart the Inciter room
+        // All groups have a chance to spawn assasins
+        case SL_BLACKHEAT_GROUP_01:
+        case SL_BLACKHEAT_GROUP_02:
+        case SL_BLACKHEAT_GROUP_03:
+        case SL_BLACKHEAT_GROUP_04:
+        case SL_BLACKHEAT_GROUP_05:
+        case SL_BLACKHEAT_GROUP_06:
+        case SL_BLACKHEAT_GROUP_07:
+        case SL_BLACKHEAT_GROUP_08:
+        case SL_BLACKHEAT_GROUP_09:
+        case SL_BLACKHEAT_GROUP_10:
+        case SL_BLACKHEAT_GROUP_11:
+        case SL_BLACKHEAT_GROUP_12:
+        {
+            switch (urand(0, 5))
```

Comment:
Chance is guessed, but after multiple tests on retail/cata i assume that every group in this room has a small chance to spawn assasins.
it was possible that killing 1 group spawned both assasins. 
There are always 
- 3 patrols (3 spawn groups)
- 2 group with 2 npc (left and right side under the tent)
- 2 static groups of 6 (1 left 1 right) 
- 2 groups with random spawns (middle left)
- 1 group with random spawns (middle right)
resulting in having 10 groups over all in this room.